test_runner/regress/test_compatibility.py

```diff
@@ -146,22 +146,29 @@ def test_create_snapshot(
     shutil.copytree(test_output_dir, compatibility_snapshot_dir)
 
 
+# TODO: this should really be a session-scoped fixture, but pg_version is function-scoped.
```

Comment:
> The problem here, is to avoid race-condition when 2 tests try to prepare the same snapshot concurrently.

Yeah I had hoped that pytest would coalesce multiple tests requesting one session-scoped fixture into a single call to instantiate the fixture.

Reply:
Going to undo the changes to `test_compatibility.py` as they're not necessary to demonstrate that this PR works.